File: services/inventory-api/item-service/src/main/java/com/inventory/item/service/ShoppingListService.java
package com.inventory.item.service;

import com.inventory.item.dto.ShoppingListItemRequest;
import com.inventory.item.dto.ShoppingListItemResponse;
import com.inventory.item.model.ShoppingListItemStatus;

import java.util.List;

public interface ShoppingListService {
    List<ShoppingListItemResponse> getShoppingList(ShoppingListItemStatus status);

    List<ShoppingListItemResponse> generateFromLowStock();

    ShoppingListItemResponse createItem(ShoppingListItemRequest request);

    ShoppingListItemResponse updateItem(Long id, ShoppingListItemRequest request);

    void deleteItem(Long id);
}

File: frontend/src/app/services/shopping-list.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpParams } from '@angular/common/http';
import { BaseApiService } from './base-api.service';
import { ShoppingListItemRequest, ShoppingListItemResponse } from '../models/item.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ShoppingListService extends BaseApiService<ShoppingListItemResponse> {
  protected override baseUrl = `${environment.apiUrl}/shopping-list`;

  getList(status?: 'PENDING' | 'PURCHASED' | 'SKIPPED'): Observable<ShoppingListItemResponse[]> {
    let params = new HttpParams();
    if (status) params = params.set('status', status);
    return this.getAll(params);
  }

  generateFromLowStock(): Observable<ShoppingListItemResponse[]> {
    return this.http.post<ShoppingListItemResponse[]>(`${this.baseUrl}/generate-from-low-stock`, {});
  }

  override create(payload: Partial<ShoppingListItemResponse> | ShoppingListItemRequest): Observable<ShoppingListItemResponse> {
    return this.http.post<ShoppingListItemResponse>(this.baseUrl, payload);
  }

  override update(id: number, payload: Partial<ShoppingListItemResponse> | ShoppingListItemRequest): Observable<ShoppingListItemResponse> {
    return this.http.patch<ShoppingListItemResponse>(`${this.baseUrl}/${id}`, payload);
  }

  delete(id: number): Observable<void> {
    return this.remove(id);
  }
}
